### PR TITLE
Style cleanups

### PR DESCRIPTION
**Pull request by eli-schwartz**
miscellaneous collection of pylint fixes, f-string upgrades, and manual nits. Should be fairly tame.

f67d0c632f4075543b99eeba351e4200a682777c is an actual bug, albeit a very minor one.

### CODE REVIEW

**Comment by eli-schwartz** [comment]
Perils of waiting while someone else adds more code that uses an import. :D

File: mesonbuild/modules/gnome.py [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ def mkenums_simple(self, state, args, kwargs):
             fhead += '%s\n' % body_prefix
         fhead += '#include "%s"\n' % hdr_filename
         for hdr in sources:
-            fhead += '#include "%s"\n' % os.path.basename(str(hdr))
+            fhead += '#include "{}"\n'.format(os.path.basename(str(hdr)))
```
Comment (dcbaker) [inline mesonbuild/modules/gnome.py:1521]: This could be an f-string, right?

**Review by dcbaker — APPROVED** [review]

File: mesonbuild/modules/gnome.py (hunk 1 above)
Comment (eli-schwartz) [reply mesonbuild/modules/gnome.py:1521]: Many things could be f-strings, but I exercise my highly opinionated judgment in deciding when stuff is "too complicated" to look good in an f-string. :D

**Comment by eli-schwartz** [comment]
Erm, the last two pushes were a mistake, accidentally pushed to the wrong branch :o

**Comment by eli-schwartz** [comment]
Also -- it did pass CI before the accidental push, so... merged.

File: mesonbuild/mintro.py [hunk 2]
```diff
@@ -492,14 +492,15 @@ def run(options: argparse.Namespace) -> int:
 
 def write_intro_info(intro_info: T.Sequence[T.Tuple[str, T.Union[dict, T.List[T.Any]]]], info_dir: str) -> None:
     global updated_introspection_files
-    for i in intro_info:
-        out_file = os.path.join(info_dir, 'intro-{}.json'.format(i[0]))
+    for kind, data in intro_info:
+        print(f'processing {kind}, {data}')
```
Comment (eli-schwartz) [inline mesonbuild/mintro.py:496]: I am dumb (and yes this was there the whole time).

**Pull request merged**